src/SDKs/Azure.Base/data-plane/Azure.Base/Http/HeaderValues.cs

```diff
@@ -0,0 +1,517 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Azure.Base.Http
+{
+    public readonly struct HeaderValues : IList<string>, IReadOnlyList<string>, IEquatable<HeaderValues>, IEquatable<string>, IEquatable<string[]>
+    {
+        private static readonly string[] EmptyArray = new string[0];
+        public static readonly HeaderValues Empty = new HeaderValues(EmptyArray);
+
+        private readonly string _value;
```

Comment:
There appears to be some inconsistency in which of these has precedence.   Some methods, such as `Count`and `GetArray` value prefer `_value` and consider `_values` only when `_value` is `null`.    Other methods, such as `GetStringValue` and `IndexOf` base their decision on the inverse of that.   Perhaps I'm missing something, but it seems that the order of precedence should be consistent.